src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,608 @@
+
+    /**
+     * Returns the factory map, lazily constructing map on first call.
+     *
+     * @return Map of RandomGeneratorFactory classes.
+     */
+    private static Map<String, Provider<? extends RandomGenerator>> getFactoryMap() {
+        return FactoryMapHolder.FACTORY_MAP;
+    }
+
+    /**
+     * Return the annotation for the specified provider.
+     *
+     * @return RandomGeneratorProperties annotation for the specified provider.
+     */
+     private RandomGeneratorProperties getProperties() {
+        if (properties == null) {
+            synchronized (provider) {
+                if (properties == null) {
+                    properties = provider.type().getDeclaredAnnotation(RandomGeneratorProperties.class);
+                    Objects.requireNonNull(properties, provider.type() + " missing annotation");
+                }
+            }
+        }
+
+        return properties;
+    }
+
+    /**
+     * Return true if the provider is a subclass of the category.
+     *
+     * @param category Interface category, sub-interface of {@link RandomGenerator}.
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private boolean isSubclass(Class<? extends RandomGenerator> category) {
+        return isSubclass(category, provider);
+    }
+
+    /**
+     * Return true if the provider is a subclass of the category.
+     *
+     * @param category Interface category, sub-interface of {@link RandomGenerator}.
+     * @param provider Provider that is being filtered.
+     *
+     * @return true if the provider is a subclass of the category.
+     */
+    private static boolean isSubclass(Class<? extends RandomGenerator> category,
+                                      Provider<? extends RandomGenerator> provider) {
+        return provider != null && category.isAssignableFrom(provider.type());
+    }
+
+    /**
+     * Returns the provider matching name and category.
+     *
+     * @param name      Name of RandomGenerator
+     * @param category  Interface category, sub-interface of {@link RandomGenerator}.
+     *
+     * @return A provider matching name and category.
+     *
+     * @throws IllegalArgumentException if provider is not a subclass of category.
+     */
+    private static Provider<? extends RandomGenerator> findProvider(String name,
+                                                                    Class<? extends RandomGenerator> category)
+            throws IllegalArgumentException {
+        Map<String, Provider<? extends RandomGenerator>> fm = getFactoryMap();
+        Provider<? extends RandomGenerator> provider = fm.get(name);
+        if (!isSubclass(category, provider)) {
+            throw new IllegalArgumentException(name + " is an unknown random number generator");
+        }
+        return provider;
+    }
+
+    /**
+     * Returns a {@link RandomGenerator} that utilizes the {@code name}
+     * <a href="package-summary.html#algorithms">algorithm</a>.
+     *
```

Comment:
In javadoc, this seems like is should read as: "utilizes the named algorithm".
The use of the parameter name is unnecessary in this case because it is obvious and readability suffers as is.

Reply:
Modified.